recipe/schemas.py

```diff
@@ -502,24 +278,81 @@ def _full_condition_schema():
         'Dimension':
             S.Dict(
                 allow_unknown=True,
+                coerce=_move_extra_fields,
                 schema={
                     'field':
                         'non_aggregated_field',
+                    'extra_fields':
+                        S.List(
+                            required=False,
+                            schema=S.Dict(
+                                schema={
+                                    'field': 'non_aggregated_field',
+                                    'name': S.String(required=True)
+                                }
+                            )
+                        ),
                     'format':
                         S.String(
                             coerce=lambda v: format_lookup.get(v, v),
                             required=False
                         )
                 },
+            ),
+        'Filter':
+            S.Dict(allow_unknown=True, schema={
+                'condition': 'condition'
+            }),
+        'Having':
+            S.Dict(
+                allow_unknown=True, schema={
+                    'condition': 'having_condition'
+                }
             )
     },
+    # If the kind can't be find, default to metric
+    default_choice=S.Dict(
```

Comment:
`default_choice` is supposed to be a string (matching one of the possible values of your key), not a schema. I'm not sure how this code is working D: We may be missing a unit test for this behavior.

This should just be `default_choice='Metric'`.

Reply:
Fixed default_chocie and the PR is merged.